python/parquet_tools/common/iofiles.py

```diff
@@ -0,0 +1,316 @@
+import os
+
+from ..common.errhandlers import ErrInfo, FatalError, ParmError
+
+
+# *****************************************************************************
+# *                               I O F i l e s                               *
+# *****************************************************************************
+
+class IOFiles:
+
+    # File information
+    #
+    fIN = []          # List of files to process
+    fOUT = []         # List of files to produce (1-to-1 corespondence)
+
+    # get_filepair() processing options
+    #
+    _blab = False
+    _skip = False
+    _skipRep = False
+    _replace = False
+
+    # *************************************************************************
+    # *                      _a p p l y _ t e m p l a t e                     *
+    # *************************************************************************
+    @classmethod
+    def _apply_template(cls, tmplt):
+        """Generate output file names from a template.
+
+        Parameters
+        ----------
+        tmplt : 'string'
+            The template to use to generate output file names (see notes).
+
+        Raises
+        ------
+        ParmError
+
+        Notes
+        -----
+        - The template is of the form '[=dir/][-sfx][+sfx]'. Where:
+            '=dir/' replaces the input file directory path with 'dir'
+            '-sfx'  removes 'sfx' from the end of the input file name.
+            '+sfx'  adds 'sfx' to the end of the input file name.
+        - The input file names are contained in our fIN list and the
+          correspodning output files names are placed in our fOUT list
+          in 1-to-1 order.
+        """
+
+        # Get optional directory
+        #
+        stmp = tmplt
+        xdir = ''
+        if tmplt[0:1] == '=':
+            n = tmplt.rfind('/')
+            if n < 0:
+                raise FatalError(3, 'directory specification in template',
+                                 stmp)
+            xdir = tmplt[1:n+1]
+            if n < len(tmplt)-1:
+                tmplt = tmplt[n+1:]
+            else:
+                tmplt = ''
+
+        # Get suffixes
+        #
+        dsfx = ''
+        asfx = ''
+        if tmplt:
+            if tmplt[0:1] == '-' and len(tmplt) > 1:
+                n = tmplt.find('+')
+                if n < 0:
+                    dsfx = tmplt[1:]
+                    tmplt = ''
+                else:
+                    dsfx = tmplt[1:n]
+                    tmplt = tmplt[n:]
+            if tmplt[0:1] == '+' and len(tmplt) > 1:
+                asfx = tmplt[1:]
+            else:
+                if tmplt:
+                    raise ParmError(3, 'suffix specification in template',
+                                    stmp)
+
+        # Apply template to generate output file names
+        #
+        for infile in IOFiles.fIN:
+            if xdir:
+                n = infile.rfind('/')
+                if n >= 0:
+                    infile = xdir+infile[n+1:]
```

Comment:
normally there would be spaces between the variable and the math symbol. (throughout the file)